src/Extensions/Hosting.Services.Web/Middlewares/ObsoleteCorrelationHeadersMiddleware.cs

```diff
@@ -89,6 +95,25 @@ private void ExtractCorrelationFromRequest(HttpRequest request)
 				? value.FirstOrDefault(s => !string.IsNullOrWhiteSpace(s))
 				: null;
 
+		private static string? ExtractHeader(IHeaderDictionary headers, IEnumerable<string> names)
+		{
+			foreach (string name in names)
+			{
+				string? value = ExtractHeader(headers, name);
```

Comment:
You can replace the ExtractHeader function with
`headers[name]`

When accessing a key that is not present in an IHeaderDictionary, StringValues.Empty is returned.